pkg/storegateway/metadata_fetcher_filters.go

```diff
@@ -75,3 +76,33 @@ func (f *IgnoreDeletionMarkFilter) FilterWithBucketIndex(_ context.Context, meta
 
 	return nil
 }
+
+func NewIgnoreNonQueryableBlocksFilter(logger log.Logger, ignoreWithin time.Duration) *IgnoreNonQueryableBlocksFilter {
+	return &IgnoreNonQueryableBlocksFilter{
+		logger:       logger,
+		ignoreWithin: ignoreWithin,
+	}
+}
+
+// IgnoreNonQueryableBlocksFilter ignores blocks that are too new be queried.
+// This has be used in conjuction with `-querier.query-store-after` with some buffer.
+type IgnoreNonQueryableBlocksFilter struct {
+	// Blocks that were created within now() - ignoreWithin will not be added to the index.
+	ignoreWithin time.Duration
+	logger       log.Logger
+	now          *time.Time
+}
+
+// Filter implements block.MetadataFilter.
+func (f *IgnoreNonQueryableBlocksFilter) Filter(ctx context.Context, metas map[ulid.ULID]*metadata.Meta, synced block.GaugeVec, modified block.GaugeVec) error {
+	ignoreWithin := time.Now().Add(-f.ignoreWithin).UnixMilli()
+
+	for id, m := range metas {
+		if m.MinTime > ignoreWithin {
```

Comment:
Min or Max? 

Hum.. seems max could make more sense?

Reply:
Ok.. never mind.. i think `min` is safer... as max will keep changing when compacting the blocks?

Reply:
There is also the possibility of a block with `MinTime < ignoreWithin && MaxTime > ignoreWithin`. This case we should sync the block because it has samples with timestamp > ignoreWithin.